### PR TITLE
Fix missing shutdown of the batch processor in otel example

### PR DESCRIPTION
**Pull request by huikang**
- Address #1183 

Signed-off-by: Hui Kang <kangh@us.ibm.com>

### CODE REVIEW

File: example/otel-collector/main.go [hunk 1]
```diff
@@ -76,17 +77,17 @@ func initProvider() (*otlp.Exporter, *push.Controller) {
 	global.SetMeterProvider(pusher.Provider())
 	pusher.Start()
 
-	return exp, pusher
+	return func() {
+		bsp.Shutdown() // shutdown the processor
+		pusher.Stop()  // pushes any last exports to the receiver
```
Comment (Aneurysm9) [inline example/otel-collector/main.go:82]: This probably also needs a call to `exp.Shutdown(context.Background())`

Comment (huikang) [reply example/otel-collector/main.go:82]: Thanks for your comment. I am wondering if there is any order in shutting down exporter and processor? I read the doc, but could not find anything.

Comment (Aneurysm9) [reply example/otel-collector/main.go:82]: That's a really good question.  My initial thought is that we'd want to shut down the processor first to ensure that its queue is drained and then shut down the exporter to ensure that any remaining spans are exported.

Comment (huikang) [reply example/otel-collector/main.go]: Updated. Please have a look. Thanks.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**